### PR TITLE
Add Firebase Functions hosting guide (#631)

### PR DESCRIPTION
**Pull request by roj1512**

### CODE REVIEW

File: site/docs/id/hosting/firebase.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+## Persiapan
+
+Pada bagian ini, kami akan memandu kamu melalui proses persiapan yang diperlukan.
+Jika kamu memerlukan penjelasan yang lebih detail untuk setiap langkah-langkahnya, silahkan lihat [dokumentasi resmi Firebase](https://firebase.google.com/docs/functions/get-started).
+
+### Membuat Proyek Firebase
+
+1. Buka [Firebase console](https://console.firebase.google.com/) lalu klik **Add Project** untuk menambahkan project baru.
+2. Jika diminta, tinjau dan setujui syarat serta kebijakan Firebase.
```
Comment (fadzikri) [inline site/docs/id/hosting/firebase.md:18]:
```suggestion
2. Jika diminta tentang kebijakan Firebase, tinjau dan setujuilah syarat tersebut.
```

Comment (fadzikri) [reply site/docs/id/hosting/firebase.md:18]: just suggestion

File: site/docs/id/hosting/firebase.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+const bot = new Bot(""); // <-- taruh token bot-mu di antara ""
+
+bot.command(
+  "start",
+  (ctx) => ctx.reply("Selamat datang! Bot berhasil dijalankan."),
+);
+bot.command("ping", (ctx) => ctx.reply(`Pong! ${new Date()}`));
+
+// Selama mengembangkan bot, kamu bisa men-trigger function kamu melalui https://localhost/<firebase-namaproyek>/us-central1/helloworld
```
Comment (fadzikri) [inline site/docs/id/hosting/firebase.md:69]:
```
// Selama mengembangkan bot, kamu bisa men-trigger function kamu melalui https://localhost/<firebase-namaproyek>/us-central1/helloworld
```

Is the `helloworld` javascript file?

Comment (fadzikri) [inline site/docs/id/hosting/firebase.md:69]:
```suggestion
// Selama mengembangkan bot, kamu bisa men-trigger function kamu melalui https://localhost/<firebase-namaproyek>/us-central1/helloWorld
```

Yes?

**Review by fadzikri — CHANGES_REQUESTED** [review]

File: site/docs/id/hosting/firebase.md (hunk 1 above)
Comment (CikiMomogi) [reply site/docs/id/hosting/firebase.md:18]:
```suggestion
2. Setujui syarat dan ketentuan Firebase yang berlaku.
```

File: site/docs/id/hosting/firebase.md (hunk 2 above)
Comment (CikiMomogi) [reply site/docs/id/hosting/firebase.md:69]: I am not sure. Perhaps @gertminov could elaborate.

File: site/docs/hosting/firebase.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+```ts
+import * as functions from "firebase-functions";
+import { Bot, webhookCallback } from "grammy";
+
+const bot = new Bot("");
+
+bot.command("start", (ctx) => ctx.reply("Welcome! Up and running."));
+bot.command("ping", (ctx) => ctx.reply(`Pong! ${new Date()}`));
+
+// During development, you can trigger your function from https://localhost/<firebase-projectname>/us-central1/helloworld
```
Comment (fadzikri) [inline site/docs/hosting/firebase.md:66]:
```suggestion
// During development, you can trigger your function from https://localhost/<firebase-projectname>/us-central1/helloWorld
```

File: site/docs/hosting/firebase.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+The Firebase CLI will give you the URL of your function once the deployment is completed.
+It should look something like `https://<REGION>.<MY_PROJECT.cloudfunctions.net/helloWorld`.
+For a more detailed explanation you can take a look at step 8. of the [get started guide](https://firebase.google.com/docs/functions/get-started#deploy-functions-to-a-production-environment).
+
+After deploying, you need to tell Telegram where to send updates to your bot by calling the `setWebhook` method.
+To do this, open a new browser tab and visit this URL:
+
+```text:no-line-numbers
+https://api.telegram.org/bot<BOT_TOKEN>/setWebhook?url=https://<REGION>.<MY_PROJECT>.cloudfunctions.net/helloworld
```
Comment (fadzikri) [inline site/docs/hosting/firebase.md:164]:
```suggestion
https://api.telegram.org/bot<BOT_TOKEN>/setWebhook?url=https://<REGION>.<MY_PROJECT>.cloudfunctions.net/helloWorld
```

File: site/docs/id/hosting/firebase.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+Firecase CLI akan memberimu URL function-nya ketika proses deployment selesai dijalankan.
+URL-nya kurang lebih terlihat seperti ini: `https://<REGION>.<MY_PROJECT>.cloudfunctions.net/helloWorld`.
+Untuk penjelasan lebih lanjut, kamu bisa melihat langkah ke-8 di [panduan memulai berikut](https://firebase.google.com/docs/functions/get-started#deploy-functions-to-a-production-environment).
+
+Setelah di-deploy, kamu perlu memberi tahu Telegram untuk mengirim update ke bot kamu dengan cara memanggil method `setWebhook`.
+Buka sebuah tab browser baru, lalu kunjungi URL berikut:
+
+```text:no-line-numbers
+https://api.telegram.org/bot<BOT_TOKEN>/setWebhook?url=https://<REGION>.<MY_PROJECT>.cloudfunctions.net/helloworld
```
Comment (fadzikri) [inline site/docs/id/hosting/firebase.md:167]:
```suggestion
https://api.telegram.org/bot<BOT_TOKEN>/setWebhook?url=https://<REGION>.<MY_PROJECT>.cloudfunctions.net/helloWorld
```

File: site/docs/zh/hosting/firebase.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+```ts
+import * as functions from "firebase-functions";
+import { Bot, webhookCallback } from "grammy";
+
+const bot = new Bot("");
+
+bot.command("start", (ctx) => ctx.reply("Welcome! Up and running."));
+bot.command("ping", (ctx) => ctx.reply(`Pong! ${new Date()}`));
+
+// 在开发期间，你可以从 https://localhost/<firebase-projectname>/us-central1/helloworld 触发你的函数。
```
Comment (fadzikri) [inline site/docs/zh/hosting/firebase.md:66]:
```suggestion
// 在开发期间，你可以从 https://localhost/<firebase-projectname>/us-central1/helloWorld 触发你的函数。
```

File: site/docs/zh/hosting/firebase.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+
+部署完成后，Firebase CLI 将为你提供函数的 URL。
+它看上去应该类似于 `https://<REGION>.<MY_PROJECT.cloudfunctions.net/helloWorld`。
+有关更详细的说明，你可以查看 [入门指南](https://firebase.google.com/docs/functions/get-started#deploy-functions-to-a-production-环境) 中的步骤 8。
+
+部署后，你需要通过调用 `setWebhook` 方法告诉 Telegram 将 update 发送到你的 bot。
+为此，请打开一个新的浏览器选项卡并访问此 URL：
+
+```text:no-line-numbers
+https://api.telegram.org/bot<BOT_TOKEN>/setWebhook?url=https://<REGION>.<MY_PROJECT>.cloudfunctions.net/helloworld
```
Comment (fadzikri) [inline site/docs/zh/hosting/firebase.md:164]:
```suggestion
https://api.telegram.org/bot<BOT_TOKEN>/setWebhook?url=https://<REGION>.<MY_PROJECT>.cloudfunctions.net/helloWorld
```

**Review by fadzikri — CHANGES_REQUESTED** [review]
Capitalize `World`.

**Review by fadzikri — APPROVED** [review]

**Comment by fadzikri** [comment]
Does it really need to be `helloWorld`?

**Comment by gertminov** [comment]
> Does it really need to be `helloWorld`? I am worried that my actions are not correct.

Yeah `helloWorld` is correct. I tested it again and can confirm, that capitalized is correct.

**Review by habemuscode — APPROVED** [review]

File: site/docs/hosting/firebase.md [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+
+To deploy your function, just run:
+
+```sh
+firebase deploy
+```
+
+The Firebase CLI will give you the URL of your function once the deployment is completed.
+It should look something like `https://<REGION>.<MY_PROJECT.cloudfunctions.net/helloWorld`.
+For a more detailed explanation you can take a look at step 8. of the [get started guide](https://firebase.google.com/docs/functions/get-started#deploy-functions-to-a-production-environment).
```
Comment (roj1512) [inline site/docs/hosting/firebase.md:158]:
```suggestion
For a more detailed explanation you can take a look at step 8 of the [get started guide](https://firebase.google.com/docs/functions/get-started#deploy-functions-to-a-production-environment).
```

**Review by CikiMomogi — APPROVED** [review]

**Pull request merged**